src/gsy_e/models/market/grid_fees/constant_grid_fees.py

```diff
@@ -47,50 +47,35 @@ def update_forwarded_bid_with_fee(self, source_rate, original_rate):
-        return trade_offer_info
-
-    def update_forwarded_offer_trade_original_info(self, trade_original_info, market_offer):
+            return None, None, None
+        if trade_original_info.propagated_bid_rate:
+            # When DoF is active, read the rate used in the trade market adding the current
+            # market's grid fee
+            propagated_bid_rate = trade_original_info.propagated_bid_rate + self.grid_fee_rate
+        else:
+            propagated_bid_rate = market_bid.energy_rate + self.grid_fee_rate
+        return (
+            trade_original_info.original_bid_rate, propagated_bid_rate,
+            trade_original_info.trade_rate)
+
+    def adapt_offer_fees_on_offer_trade(self, trade_original_info, market_offer):
         if not trade_original_info:
-            return None
-        trade_bid_info = TradeBidOfferInfo(
-            original_bid_rate=trade_original_info.original_bid_rate,
-            propagated_bid_rate=trade_original_info.propagated_bid_rate,
-            original_offer_rate=market_offer.original_price / market_offer.energy,
-            propagated_offer_rate=market_offer.energy_rate,
-            trade_rate=trade_original_info.trade_rate)
-        return trade_bid_info
-
-    def propagate_original_bid_info_on_offer_trade(self, trade_original_info):
+            return None, None, None
+        original_offer_rate = market_offer.original_price / market_offer.energy
+        return original_offer_rate, market_offer.energy_rate, trade_original_info.trade_rate
+
+    def adapt_bid_fees_on_offer_trade(self, trade_original_info):
         if trade_original_info is None:
-            return None
+            return None, None, None
         bid_rate = trade_original_info.propagated_bid_rate - self.grid_fee_rate
-        trade_bid_info = TradeBidOfferInfo(
-            original_bid_rate=trade_original_info.original_bid_rate,
-            propagated_bid_rate=bid_rate,
-            original_offer_rate=None,
-            propagated_offer_rate=None,
-            trade_rate=trade_original_info.trade_rate)
-        return trade_bid_info
-
-    def propagate_original_offer_info_on_bid_trade(self, trade_original_info, ignore_fees=False):
+        return trade_original_info.original_bid_rate, bid_rate, trade_original_info.trade_rate
+
+    def adapt_offer_fees_on_bid_trade(self, trade_original_info, ignore_fees=False):
         grid_fee_rate = self.grid_fee_rate if not ignore_fees else 0.0
         offer_rate = trade_original_info.propagated_offer_rate + grid_fee_rate
-        trade_offer_info = TradeBidOfferInfo(
-            original_bid_rate=None,
-            propagated_bid_rate=None,
-            original_offer_rate=trade_original_info.original_offer_rate,
-            propagated_offer_rate=offer_rate,
-            trade_rate=trade_original_info.trade_rate)
-        return trade_offer_info
+        return trade_original_info.original_offer_rate, offer_rate, trade_original_info.trade_rate
```

Comment:
The last element of the return tuple seems to be never used in the calls of `adapt_offer_fees_on_bid_trade`. IMO you can skip returning it.

Reply:
You are right, we're not currently using it but I left it as this just in order to homogenize the `adapt_offer_fees_on_bid_trade` and `adapt_bid_fees_on_bid_trade` return value